atc/api/idtokenserver/server.go

```diff
@@ -10,17 +10,26 @@ import (
 type Server struct {
 	logger              lager.Logger
 	externalURL         string
+	oidcIssuer          string
 	dbSigningKeyFactory db.SigningKeyFactory
 }
 
 func NewServer(
 	logger lager.Logger,
 	externalURL string,
+	oidcIssuer string,
```

Comment:
How about only passing oidcIssuer here and it either contains the value of the flag or of externalURL.
The server does not need to know where the issuer value is coming from, just what it is supposed to be